pymc_marketing/clv/models/shifted_beta_geo.py

```diff
@@ -0,0 +1,203 @@
+
+        coords = {"customer_id": np.asarray(customer_id)}
+        with pm.Model(coords=coords) as self.model:
+            alpha = self.model.register_rv(alpha_prior, name="alpha")
+            beta = self.model.register_rv(beta_prior, name="beta")
+
+            theta = pm.Beta("theta", alpha, beta, dims=("customer_id",))
+
+            churn_raw = pm.Geometric.dist(theta)
+
+            pm.Censored(
+                "churn_censored",
+                churn_raw,
+                lower=None,
+                upper=T,
+                observed=t_churn,
+                dims=("customer_id",),
+            )
+
+    def _process_priors(self, alpha_prior, beta_prior):
+        if alpha_prior is None:
+            alpha_prior = pm.HalfFlat.dist()
+        else:
+            assert alpha_prior.eval().shape == ()
+        if beta_prior is None:
+            beta_prior = pm.HalfFlat.dist()
+        else:
+            assert beta_prior.eval().shape == ()
+
+        assert alpha_prior is not beta_prior
+
+        alpha_prior.str_repr = types.MethodType(str_for_dist, alpha_prior)
+        beta_prior.str_repr = types.MethodType(str_for_dist, beta_prior)
+
+        return alpha_prior, beta_prior
+
+    def distribution_customer_churn_time(
+        self, customer_id: Union[np.ndarray, pd.Series], random_seed: RandomState = None
+    ) -> DataArray:
+        """Sample distribution of churn time for existing customers
+
+        The draws represent the number of periods into the future after which
+        a customer cancels their contract.
+
+        It ignores that some customers may have already cancelled.
+        """
+
+        coords = {"customer_id": np.asarray(customer_id)}
+        with pm.Model(coords=coords):
+            alpha = pm.HalfFlat("alpha")
+            beta = pm.HalfFlat("beta")
+
+            theta = pm.Beta("theta", alpha, beta, dims=("customer_id",))
+            pm.Geometric("churn", theta, dims=("customer_id",))
+
+            return pm.sample_posterior_predictive(
+                self.fit_result,
+                var_names=["churn"],
+                random_seed=random_seed,
+            ).posterior_predictive["churn"]
+
+    def _distribution_new_customer(
+        self,
+        n: int = 1,
+        random_seed: RandomState = None,
+        var_names: Sequence[str] = ("theta", "churn"),
+    ) -> Dataset:
+        coords = {"new_customer_id": np.arange(n)}
+        with pm.Model(coords=coords):
+            alpha = pm.HalfFlat("alpha")
+            beta = pm.HalfFlat("beta")
+
+            theta = pm.Beta("theta", alpha, beta, dims=("new_customer_id",))
+            pm.Geometric("churn", theta, dims=("new_customer_id",))
+
+            return pm.sample_posterior_predictive(
+                self.fit_result,
+                var_names=var_names,
+                random_seed=random_seed,
+            ).posterior_predictive
+
+    def distribution_new_customer_churn_time(
```

Comment:
Can we add docstring so that we understand what `n` means (again, thinking about a user) as this is a public method?

Reply:
(same for all public methods)

Reply:
Yes, my bad. I thought I already added those before.

Reply:
Done